### DISCUSSION TITLE
Depth buffer Texture never removed

### DISCUSSION THREAD

**Discussion by inveders** [google/filament#4616] [Q&A] [answered]
Hi, this question is relative to [this](https://github.com/google/filament/discussions/4210) concerning the hole in the rectangle. All is fine, I draw my invisible texture in the depth buffer, I have my hole, but when I want to erase all the screen to reset it, I can't remove the renderable wich have the depth buffer texture. So when I re-create an other wall after, I always have the hole persistent on my screen (so when I draw the new wall, it have the last hole on it). Is it a way to clear or remove the renderable drawn in the depth buffer? I don't know if my question is understandable?
Thanks you !!

**Comment by romainguy** [comment, marked as answer]
You cannot clear the depth buffer partly through the frame no (and it would be inefficient/lead to rendering bugs most likely).

**Reply by inveders** [reply]
Hum ok, so no solution to reset all? Am I forced to kill and re-start the ARCore session to reset depth buffer in Filament?

**Reply by romainguy** [reply]
To be clear, the depth buffer is cleared at the beginning of *each* frame. So if you remove your `Renderable` from the scene (or change its layer mask to make it invisible), it won't render again. You just can't clear the depth buffer or "remove" something from the depth buffer halfway through the frame. The depth buffer has *nothing* to do with the ARCore session.

**Reply by inveders** [reply]
Oh ok I understand, thanks you !!
